src/crypto/isa-l/CMakeLists.txt

```diff
@@ -1,36 +1,44 @@
-set(isal_dir ${CMAKE_SOURCE_DIR}/src/crypto/isa-l/isa-l_crypto)
-set(CMAKE_ASM_FLAGS "-i ${isal_dir}/aes/ -i ${isal_dir}/include/ ${CMAKE_ASM_FLAGS}")
+set(ISAL_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/isa-l_crypto)
+set(ISAL_INCLUDE_DIR "${ISAL_SOURCE_DIR}/include")
+set(ISAL_LIBRARY "${ISAL_SOURCE_DIR}/bin/isa-l_crypto.a")
+
+include(FindMake)
+find_make("MAKE_EXECUTABLE" "make_cmd")
+
+# use an ExternalProject to build isa-l_crypto using its makefile
+include(ExternalProject)
+ExternalProject_Add(isal_crypto_ext
+  SOURCE_DIR ${ISAL_SOURCE_DIR}
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ${make_cmd} -f <SOURCE_DIR>/Makefile.unx
+  BUILD_IN_SOURCE 1
+  BUILD_BYPRODUCTS ${ISAL_LIBRARY}
+  INSTALL_COMMAND ""
+  UPDATE_COMMAND ""
+  LOG_CONFIGURE ON
+  LOG_BUILD ON
+  LOG_MERGED_STDOUTERR ON
+  LOG_OUTPUT_ON_FAILURE ON)
+
+# add imported library target ISAL::Crypto
+add_library(ISAL::Crypto STATIC IMPORTED GLOBAL)
+add_dependencies(ISAL::Crypto isal_crypto_ext)
+set_target_properties(ISAL::Crypto PROPERTIES
+  INTERFACE_INCLUDE_DIRECTORIES ${ISAL_INCLUDE_DIR}
+  IMPORTED_LINK_INTERFACE_LANGUAGES "C"
+  IMPORTED_LOCATION ${ISAL_LIBRARY})
```

Comment:
Hi @cbodley  These contents should be formed into a separate file `Buildisal.cmake` and placed in the `ceph/cmake/modules/` directory.